Minesweeper/Minesweeper/Entities/CollectionExtension.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  CollectionExtension.swift
+//  Minesweeper
+//
+//  Created by Dorian Grolaux on 21/02/2018.
+//
+
+import Foundation
+
+extension Collection {
+    
+    /// Returns the element at the specified index iff it is within bounds, otherwise nil.
+    subscript (safe index: Index) -> Element? {
```

Comment:
When do you for-see us needing to put arguments in to the array which are out of bounds?
This seems like the sort of over protective unwrapping you once opposed. We'll be unwrapping literally every array when to the best of my knowledge we don't have a use case yet (maybe I'll read it later in this review) where we'd do anything other than force unwrap this.